### PR TITLE
Rest state

### PR DESCRIPTION
**Pull request by CalebUAz**
Completed the timer for rest state.

### CODE REVIEW

File: run_client.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from tasks.affective_task import ClientAffectiveTask
 from tasks.finger_tapping_task import ClientFingerTappingTask
 from tasks.ping_pong_task import ClientPingPongTask
+from tasks.rest_state import ClientRestState, client_rest_state
```
Comment (eduongAZ) [inline run_client.py:19]: You should not import `client_rest_state` from `tasks.rest_state` module

File: run_server.py [hunk 2]
```diff
@@ -37,6 +39,16 @@ def _run_ping_pong(to_client_connections: list, from_client_connections: dict, s
...
     server = Server(args.address, args.port)
 
+    # Initial rest state
+
+    server.establish_connections(REQUIRED_NUM_CONNECTIONS_REST_STATE)
+
+    _send_start(list(server.to_client_connections.values()))
+
+    server_rest_state = ServerRestState(list(server.to_client_connections.values()), 
+                                                         server.from_client_connections)
```
Comment (eduongAZ) [inline run_server.py:49]: Keep indentation consistent

File: tasks/rest_state/__init__.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+from .client_rest_state import ClientRestState
+from .server_rest_state import ServerRestState
```
Comment (eduongAZ) [inline tasks/rest_state/__init__.py:2]: Add a newline to the end of file

File: tasks/rest_state/client_rest_state.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+            # show a blank screen and a cross before showing an image
+            render_blank_screen(self._screen, BLANK_SCREEN_MILLISECONDS)
+
+            wait(CROSS_SCREEN_MILLISECONDS)
+
+            timer(state["rest_timer"], [], "Please sit back, relax and try not to move for: ", self._screen)
+            response = {"type": "STOP"}
+            send([self._to_server], response)
+        print("[STATUS] Affective task ended")
```
Comment (eduongAZ) [inline tasks/rest_state/client_rest_state.py:43]: Add a newline to the end of file

File: tasks/rest_state/client_rest_state.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    def run(self):
+        print("[STATUS] Running Rest state")
+
+        while True:
+
+            [data] = receive([self._from_server])
+            print(data)
+            if data["type"] == "request":
+                if data["request"] == "end":
+                    print('im in break')
```
Comment (eduongAZ) [inline tasks/rest_state/client_rest_state.py:25]: This message should be formatted as
[INFO] Break period

File: tasks/rest_state/config_rest_state.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+BLANK_SCREEN_MILLISECONDS = 1000
+CROSS_SCREEN_MILLISECONDS = 500
+
+REST_TIMER = 30 #300000 #five minutes in milliseconds
```
Comment (eduongAZ) [inline tasks/rest_state/config_rest_state.py:4]: Add a newline to the end of file

File: tasks/rest_state/server_rest_state.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+            print("[STATUS] Running rest state")
+            send(self._to_client_connections, data)
+            
+            sleep(0.1)
+
+            while(True):
+                responses = receive(self._from_client_connections)
+                response = list(responses.values())[0]
+                if response["type"] == "STOP":
```
Comment (eduongAZ) [inline tasks/rest_state/server_rest_state.py:31]: Should make an else case with useful message to indicate that the data is invalid. This is helpful for debugging later on.

File: tasks/rest_state/utils/__init__.py [hunk 8]
```diff
@@ -0,0 +1,2 @@
+from .render_text_center import render_text_center
+from .timer import timer
```
Comment (eduongAZ) [inline tasks/rest_state/utils/__init__.py:2]: Add a newline at the end of file.

File: tasks/rest_state/utils/render_text_center.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    text_box.fill(COLOR_BACKGROUND)
+
+    text_box_shape_x, text_box_shape_y = text_box_shape
+    _, _, text_width, _ = text_render.get_rect()
+    text_box.blit(text_render, ((text_box_shape_x - text_width) / 2, 0))
+
+    center = ((CLIENT_WINDOW_WIDTH - text_box_shape_x) / 2 + x_offset, (CLIENT_WINDOW_HEIGHT - text_box_shape_y) / 2 + y_offset)
+    screen.blit(text_box, center)
+
+    pygame.display.flip()
```
Comment (eduongAZ) [inline tasks/rest_state/utils/render_text_center.py:27]: There already exists `render_text_center.py` somewhere else. Move it to `common` module if it is used by multiple modules and import it here.

**Review by eduongAZ — CHANGES_REQUESTED** [review]
Remove .DS_Store files.

File: tasks/rest_state/client_rest_state.py (hunk 5 above)
Comment (CalebUAz) [reply tasks/rest_state/client_rest_state.py:25]: I don't need these, I just used them for debugging. I'll remove these.

**Review by eduongAZ — CHANGES_REQUESTED** [review]
There are .DS_Store files that are not removed.

**Review by eduongAZ — APPROVED** [review]

**Pull request merged**